src/DynamoCore/Linting/LinterExtensionDescriptor.cs

```diff
@@ -20,5 +20,39 @@ public LinterExtensionDescriptor(string id, string name)
             Id = id;
             Name = name;
         }
+
+        /// <summary>
+        /// Checks whether two LinterExtensionDescriptor are equal
+        /// They are equal if their Name and Id are equal
+        /// </summary>
+        /// <param name="obj"></param>
+        /// <returns></returns>
+        public override bool Equals(object obj)
+        {
+            if (obj == null)
+            {
+                return false;
+            }
+            if (!(obj is LinterExtensionDescriptor))
```

Comment:
I think you could simplify this with `(if obj is LinterExtensionDescriptor other)`
https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/functional/pattern-matching#null-checks